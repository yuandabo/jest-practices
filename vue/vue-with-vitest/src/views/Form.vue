<template>
    <div class="form">
        <h1>This is an form page</h1>
        <div>
            <label htmlFor="username-input">Username</label>
            <input id="username-input" v-model="input" />
            <button @click="handleClick">
                提交
            </button>
            <span>{{ status }}</span>
        </div>
    </div>
</template>

<script setup lang="ts">
import { ref } from 'vue';

const status = ref('')
const input = ref('')
const handleClick = () => {
    const username = input.value;
    if (!username) {
        status.value = "用户名不能为空"
        return
    }
    status.value = "提交成功"
}
</script>

<style></style>
