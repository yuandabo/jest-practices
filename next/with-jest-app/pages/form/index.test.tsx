/**
 * @jest-environment jsdom
 */
import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import Form from "./index";

it("Form", () => {
  render(<Form />);
  //   expect(screen.getByRole("label")).toHaveTextContent("Username");
  // screen.getByLabelText('用户名')
  // 模拟输入
  fireEvent.change(screen.getByLabelText("Username"), {
    target: { value: "test" },
  });
  fireEvent.click(screen.getByRole("button"));
  waitFor(() => {
    expect(screen.getByText("提交成功")).toBeInTheDocument();
  });
});

test("空用户名触发错误", () => {
  const { getByRole, getByText } = render(<Form />);
  fireEvent.click(screen.getByRole("button"));
  expect(getByText("用户名不能为空")).toBeInTheDocument(); // 引用示例[7](@ref)
});

// 生成快照
test("Form快照", () => {
  const { asFragment } = render(<Form />);
  expect(asFragment()).toMatchSnapshot();
});
