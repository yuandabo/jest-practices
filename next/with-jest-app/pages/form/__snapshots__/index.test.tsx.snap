// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form快照 1`] = `
<DocumentFragment>
  <label
    for="username-input"
  >
    Username
  </label>
  <input
    id="username-input"
  />
  <button>
    提交
  </button>
  <span />
</DocumentFragment>
`;
